powershell/upload/UploadFile.ps1

```
@@ -0,0 +1,533 @@
+        $this._tenantId = $tenantId
+        $this._boundary = "file_info"
+        $this._businessTypeId = $businessTypeId
+
+        # API supports files up to 100 megabytes
+        $this._uploadSizeLimit = 100 * 1024 * 1024
+    }
+
+    [string] CreateFileToUpload([string] $contentFilePath) {
+        $headerFilePath = ""
+        $footerFilePath = ""
+        try {
+            $folderPath = $(Split-Path -Path $contentFilePath)
+            $contentFilename = $(Split-Path -Path $contentFilePath -Leaf)
+            $createdFilePath = "$($folderPath)\$([Helper]::ConverToUniqueFileName("multipart.bin"))"
+
+            $headerFilename = [Helper]::ConverToUniqueFileName("header.txt")
+            $headerFilePath = "$($folderPath)\$($headerFilename)"
+            $headerContent = "--$($this._boundary)`r`n" # Windows line breaks are required.
+            $headerContent += "Content-Type: application/json; charset=UTF-8`r`n"
+            $headerContent += "`r`n"
+            $headerContent += "{`r`n`"name`":`"$($contentFilename)`",`r`n`"businesstypeid`":`"$($this._businessTypeId)`"`r`n}`r`n"
+            $headerContent += "--$($this._boundary)`r`n`r`n"
+
+            $footerFilename = [Helper]::ConverToUniqueFileName("footer.txt")
+            $footerFilePath = "$($folderPath)\$($footerFilename)"
+            $footerContent = "`r`n--$($this._boundary)--"
+
+            New-Item -Path $folderPath -Name $headerFilename -Value $headerContent
+            New-Item -Path $folderPath -Name $footerFilename -Value $footerContent
+
+            cmd /c copy /b $headerFilePath + $contentFilePath + $footerFilePath $createdFilePath
+            return $createdFilePath
+        }
+        finally {
+            Remove-Item -Path $headerFilePath
+            Remove-Item -Path $footerFilePath
+        }
+    }
+
+    [void] UploadFile([string] $filePath) {
+        if ((Get-Item $filePath).Length -gt $this._uploadSizeLimit) {
+            Write-Host "---" -ForegroundColor "Red"
```

Good enough. But I would throw an error instead of doing the return.
throw "Cannot upload files bigger $($this._uploadSizeLimit) bytes."

---

If you don't throw the exception, change the `Write-Host "---"` to `Write-Host "----"` (added one `-`).